### ISSUE TITLE
const [a, b] = xs = [1, 2] produces wrong error: TS2304: Cannot find name 'xs'

### ISSUE THREAD

**Issue by Jendrik3** [Microsoft/TypeScript#13755]
**TypeScript Version:**  2.1.5 

**Code**

```ts
const [a, b] = xs = [1, 2];
```

**Expected behavior:**
No compile errors.

**Actual behavior:**
Compiler reports `bug.ts(1,16): error TS2304: Cannot find name 'xs'`, but emits correct javascript.

**Comment by HerringtonDarkholme** [comment]
Because it is an error. A statement with consecutive assignment like `var a = b = 123` will make a global variable `b` in sloppy mode, and an error in strict mode. Note the first `var a` is declaration, the second `b = ` is just an assignment.

Try this in node.

```ts
'use strict'
var a = b = 123
```

So you should declare `xs` first. `let xs = []; const [a, b] = xs = [1, 2];` should work.
